Zend/tests/operator_unsupported_types.phpt

```diff
@@ -0,0 +1,860 @@
+--TEST--
+Using unsupported types with operators
+--FILE--
+<?php
+
+$binops = [
+    '+',
```

Comment:
Would it be useful to have a test like this for AST_ASSIGN_OP (`+=`, `-=`, etc) to brute force all operands for that as well?

Also, I'm not familiar with the internals of `eval()` enough to know this - Does the code fragment get passed to opcache if opcache is enabled? If not, it might be useful to file_put_contents() a script with various functions, to see if the zend_may_throw() value was correct, especially with opcache's JIT.
(But that may be overkill)

```php
function test1() {
    $x = STDERR;
    $x /= 2;
    echo "Unreachable resource /= int\n";  // test fails if this is output.
}
// ... thousands of automatically generated tests
try {
    test1();
} catch(Error $e) {
    echo $e->getMessage(), "\n";
}
```